### PR TITLE
RWA-982 [SH] Upgrading springboot to 2.6.3 and gradle to 7.3.3

### PR DESCRIPTION
**Pull request by satvinder-hullait**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RWA-982

### Change description ###

Spring boot upgrade

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -315,51 +329,51 @@ dependencies {
...
-  contractTestCompile group: 'au.com.dius.pact.provider', name: 'junit5', version: versions.pactVersion
-  contractTestCompile group: 'au.com.dius.pact.provider', name: 'spring', version: versions.pactVersion
-  contractTestCompile group: 'au.com.dius.pact.provider', name: 'junit5spring', version: versions.pactVersion
+  contractTestImplementation group: 'au.com.dius.pact.provider', name: 'junit5', version: versions.pactVersion
+  contractTestImplementation group: 'au.com.dius.pact.provider', name: 'spring', version: versions.pactVersion
+  contractTestImplementation group: 'au.com.dius.pact.provider', name: 'junit5spring', version: versions.pactVersion
 
-  contractTestCompile("org.junit.jupiter:junit-jupiter-api:5.6.2")
-  contractTestRuntime("org.junit.jupiter:junit-jupiter-engine:5.6.2")
+  contractTestImplementation("org.junit.jupiter:junit-jupiter-api:5.6.2")
```
Comment (sullivangareth) [inline build.gradle:375]: why is this version different to `versions.junit`

File: src/main/java/uk/gov/hmcts/reform/waworkflowapi/controllers/startworkflow/CreateTaskController.java [hunk 2]
```diff
@@ -94,32 +94,32 @@ public CreateTaskController(EvaluateDmnService evaluateDmnService,
     }
 
     @PostMapping(path = "/workflow/message")
-    @ApiOperation("Sends a message to the underlying business process engine.")
+    @Operation(summary = "Sends a message to the underlying business process engine.")
     @ApiResponses({
```
Comment (sullivangareth) [inline src/main/java/uk/gov/hmcts/reform/waworkflowapi/controllers/startworkflow/CreateTaskController.java:98]: This uses ``@ApiResponses` but `evaluateDmn` doesn't ? Consistency ?

File: src/integrationTest/java/uk/gov/hmcts/reform/waworkflowapi/config/OpenAPIPublisherTest.java [hunk 3]
```diff
@@ -17,15 +17,15 @@
...
+class OpenAPIPublisherTest extends SpringBootIntegrationBaseTest {
 
     @Autowired
     private MockMvc mvc;
 
     @DisplayName("Generate swagger documentation")
     @Test
     void generateDocs() throws Exception {
-        byte[] specs = mvc.perform(get("/v2/api-docs"))
+        byte[] specs = mvc.perform(get("/v3/api-docs"))
```
Comment (sullivangareth) [inline src/integrationTest/java/uk/gov/hmcts/reform/waworkflowapi/config/OpenAPIPublisherTest.java:28]: Why has this version incremented ?

Comment (satvinder-hullait) [reply src/integrationTest/java/uk/gov/hmcts/reform/waworkflowapi/config/OpenAPIPublisherTest.java:28]: as part of moving from sprinfox to springdoc, we are now using OpenApi/Swagger version3

File: build.gradle (hunk 1 above)
Comment (satvinder-hullait) [reply build.gradle:375]: I've not changed the version, only changed contractTestCompile to contractTestImplementation

File: src/main/java/uk/gov/hmcts/reform/waworkflowapi/controllers/startworkflow/CreateTaskController.java (hunk 2 above)
Comment (satvinder-hullait) [reply src/main/java/uk/gov/hmcts/reform/waworkflowapi/controllers/startworkflow/CreateTaskController.java:98]: good point

**Review by sullivangareth — APPROVED** [review]

**Pull request merged**